packages/features/ee/organizations/pages/settings/general.tsx

```diff
@@ -0,0 +1,195 @@
+import { useRouter } from "next/router";
+import { Controller, useForm } from "react-hook-form";
+
+import { getLayout } from "@calcom/features/settings/layouts/SettingsLayout";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { nameOfDay } from "@calcom/lib/weekday";
+import { MembershipRole } from "@calcom/prisma/enums";
+import type { RouterOutputs } from "@calcom/trpc/react";
+import { trpc } from "@calcom/trpc/react";
+import {
+  Button,
+  Form,
+  Label,
+  Meta,
+  Select,
+  showToast,
+  SkeletonButton,
+  SkeletonContainer,
+  SkeletonText,
+  TimezoneSelect,
+} from "@calcom/ui";
+
+const SkeletonLoader = ({ title, description }: { title: string; description: string }) => {
+  return (
+    <SkeletonContainer>
+      <Meta title={title} description={description} />
+      <div className="mt-6 mb-8 space-y-6">
+        <SkeletonText className="h-8 w-full" />
+        <SkeletonText className="h-8 w-full" />
+        <SkeletonText className="h-8 w-full" />
+        <SkeletonText className="h-8 w-full" />
+
+        <SkeletonButton className="mr-6 h-8 w-20 rounded-md p-5" />
+      </div>
+    </SkeletonContainer>
+  );
+};
+
+interface GeneralViewProps {
+  currentOrg: RouterOutputs["viewer"]["organizations"]["listCurrent"];
+  isAdminOrOwner: boolean;
+  localeProp: string;
+}
+
+const OrgGeneralView = () => {
+  const { t } = useLocale();
+  const router = useRouter();
+
+  const { data: currentOrg, isLoading } = trpc.viewer.organizations.listCurrent.useQuery(undefined, {
+    onError: () => {
+      router.push("/settings");
+    },
+  });
+  const { data: user } = trpc.viewer.me.useQuery();
+
+  if (isLoading) return <SkeletonLoader title={t("general")} description={t("general_description")} />;
+  if (!currentOrg) {
+    throw new Error(t("something_went_wrong"));
+  }
```

Comment:
Should we really throw an Error here IIRC this would cause a 500 on the client - Should we not redirect (as they shouldnt be in an org) ? We can do a check in the `onSettled` and redirect back to settings?

Reply:
I have already added    
```
onError: () => {
      router.push("/settings");
}, 
```
in trpc.viewer.organizations.listCurrent

Reply:
```
  if (!currentOrg) {
    throw new Error(t("something_went_wrong"));
  }
  ```
  Wont count as a trpc error tho right 🤔

Reply:
yeah this is just for type checking ensuring that currentOrg exist and then passing to <GeneralView />. If there is an error while fetching or organisation doesn't exist we would be automatically redirected to /settings because of onError.

Reply:
we would have to do this a lot of currentOrg && currentOrg.timeZone.  if the condition ( if(!currentOrg) ) is not added

Reply:
Not sure i like the idea of throwing an error - maybe we just return an empty component incase something goes wrong? Also let me know if you dont like this idea :D 

 if(!currentOrg) {
return null
}